src/operators/reduce-support.ts

```diff
@@ -0,0 +1,57 @@
+import Operator from '../Operator';
+import Observer from '../Observer';
+import Subscriber from '../Subscriber';
+
+import tryCatch from '../util/tryCatch';
+import {errorObject} from '../util/errorObject';
+
+export class ReduceOperator<T, R> implements Operator<T, R> {
+
+  acc: R;
+  project: (acc: R, x: T) => R;
+
+  constructor(project: (acc: R, x: T) => R, acc?: R) {
```

Comment:
In my case I do apply small refactoring like this when I create PR, but if there's chunk of changes can be created separate PR that seems also way. Think @blesh can advise way feasible for maintaining repo history.